tests/scenario/test_deploy.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from unittest.mock import Mock, PropertyMock
+
+from charms.harness_extensions.v0.evt_sequences import Event, Scenario
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
+from ops.model import BlockedStatus, Container, WaitingStatus
+from ops.pebble import Error as PebbleError
+from ops.pebble import PathError
+
+from charm import MimirK8SOperatorCharm
+
+
+def setup_function():
+    Container.can_connect = Mock(return_value=True)
+    MimirK8SOperatorCharm._mimir_version = PropertyMock(return_value="2.4.0")
+    MimirK8SOperatorCharm._current_mimir_config = PropertyMock(return_value={})
+    MimirK8SOperatorCharm._set_alerts = Mock(return_value=True)
+    KubernetesServicePatch.__init__ = Mock(return_value=None)
+
+
+def generate_scenario():
+    return Scenario.from_events(
+        ("install", "config-changed", "start", Event("mimir-pebble-ready", workload=Mock()))
+    )(MimirK8SOperatorCharm).play_until_complete()
+
+
+def test_deploy_ok_scenario():
+    expected_plan = {
+        "services": {
+            "mimir": {
+                "override": "replace",
+                "summary": "mimir daemon",
+                "command": "/bin/mimir --config.file=/etc/mimir/mimir-config.yaml",
+                "startup": "enabled",
+            }
+        },
+    }
+    cc = generate_scenario()
+    assert cc[2].harness.get_container_pebble_plan("mimir").to_dict() == expected_plan
```

Comment:
if you want to keep these checks isolated, you could turn this function into an `assert_plan_ok` and call it from `test_startup_sequence` as part of the closing assertions.